core/src/main/java/com/datastrato/gravitino/storage/relational/mysql/MySQLBackend.java

```diff
@@ -25,41 +38,207 @@ public class MySQLBackend implements RelationalBackend {
+          case TABLE:
+          case FILESET:
+          default:
+            throw new IllegalArgumentException(
+                String.format("Unsupported entity type: %s for update operation", entityType));
+        }
+      } catch (Throwable t) {
+        SqlSessions.rollbackAndCloseSqlSession();
+        throw new RuntimeException(t);
+      }
+    }
   }
 
   @Override
   public <E extends Entity & HasIdentifier> E get(
-      NameIdentifier ident, Entity.EntityType entityType) throws IOException {
-    throw new UnsupportedOperationException("Unsupported operation now.");
+      NameIdentifier ident, Entity.EntityType entityType)
+      throws NoSuchEntityException, IOException {
+    try (SqlSession session = SqlSessions.getSqlSession()) {
+      try {
+        switch (entityType) {
+          case METALAKE:
+            MetalakePO metalakePO =
+                ((MetalakeMetaMapper) SqlSessions.getMapper(MetalakeMetaMapper.class))
+                    .selectMetalakeMetaByName(ident.name());
+            if (metalakePO == null) {
+              return null;
+            }
+            return (E) POConverters.fromMetalakePO(metalakePO);
+          case CATALOG:
+          case SCHEMA:
+          case TABLE:
+          case FILESET:
+          default:
+            throw new IllegalArgumentException(
+                String.format("Unsupported entity type: %s for get operation", entityType));
+        }
+      } catch (Throwable t) {
+        throw new RuntimeException(t);
+      } finally {
+        SqlSessions.closeSqlSession();
+      }
+    }
   }
 
   @Override
   public boolean delete(NameIdentifier ident, Entity.EntityType entityType, boolean cascade) {
-    throw new UnsupportedOperationException("Unsupported operation now.");
+    try (SqlSession session = SqlSessions.getSqlSession()) {
+      try {
+        switch (entityType) {
+          case METALAKE:
+            Long metalakeId =
+                ((MetalakeMetaMapper) SqlSessions.getMapper(MetalakeMetaMapper.class))
+                    .selectMetalakeIdMetaByName(ident.name());
+            if (metalakeId != null) {
+              // delete metalake
+              ((MetalakeMetaMapper) SqlSessions.getMapper(MetalakeMetaMapper.class))
+                  .deleteMetalakeMetaById(metalakeId);
+              if (cascade) {
+                // TODO We will cascade delete the metadata of sub-resources under the metalake
+              }
+              SqlSessions.commitAndCloseSqlSession();
+            }
+            return true;
+          case CATALOG:
+          case SCHEMA:
+          case TABLE:
+          case FILESET:
+          default:
+            throw new IllegalArgumentException(
+                String.format("Unsupported entity type: %s for delete operation", entityType));
+        }
+      } catch (Throwable t) {
+        SqlSessions.rollbackAndCloseSqlSession();
+        throw new RuntimeException(t);
+      }
+    }
   }
 
   @Override
```

Comment:
Do we need to close the `DataSource` pool?

Reply:
I mean if we are going to close `MySQLBackend`, shouldn't we close the pool at the same time.